### PR TITLE
Add an Optional Fullscreen API for Cases Where Fullscreen =/= Screen Dimens

### PR DESCRIPTION
**Pull request by daytime-em**
There are also a number of edge cases where fullscreen cannot be accurately detected. This approach allows affected users to set the fullscreen status themselves.

This also adds Fullscreen handling to the Demo app

### CODE REVIEW

**Comment by daytime-em** [comment]
This is a draft for now. Once I have feedback, I can add the Fullscreen code for other ExoPlayer versions and merge.

Thanks!

**Comment by daytime-em** [comment]
Oh, we'll also need to release MuxCore 7.0.5 and depend on the public jar before we can release the next update.

File: MuxExoPlayer/src/main/java/com/mux/stats/sdk/muxstats/MuxBaseExoPlayer.java [hunk 1]
```diff
@@ -183,26 +184,24 @@
    * @param customerPlayerData basic playback data set by the Application.
    * @param customerVideoData basic Video data set by the Application.
    * @param customerViewData basic View data set by the application.
-   * @param sentryEnabled if set to true the underlying {@link MuxStats} will report internal errors
-   *                      to the backend.
    * @param networkRequest internet interface implementation.
    */
   @Deprecated
```
Comment (jsanford8) [inline MuxExoPlayer/src/main/java/com/mux/stats/sdk/muxstats/MuxBaseExoPlayer.java:192]: I think we used to document this, or we've at least given it out to customers. I'm fine removing this as an option completely, but I think we'd need to technically major version the release.

File: build.gradle [hunk 2]
```diff
@@ -33,7 +33,7 @@ buildscript {
 //
 // It must be false for fat aar generation of MuxExoPlayer to work
 // This is a bug/problem with the fat aar mechanism
-def inLocalDevModeFlag = false
+def inLocalDevModeFlag = true
```
Comment (jsanford8) [inline build.gradle]: is this intended now?

**Review by jsanford8 — COMMENTED** [review]
in general I like it. Two quick comments.

Other than that, for when we allowed someone to set the player size, we needed to make sure that we remembered that someone overwrote it, so that we didn't then overwrite it later in our normal process. Do we want to do something similar here? I'd just be concerned that if they tell us it's fullscreen, the next time we check we'll say "oh it's not fullscreen" and toggle it back the other way.

That might not be an issue, but it's possible that we document this method with something like "by default we'll try to do our own detection of fullscreen, but if you want to control this, you can use this method. Note that if you do, it is expected that you notify the SDK of all presentation changes in the future, as this disables our automatic detection.

File: build.gradle (hunk 2 above)
Comment (skidder) [reply build.gradle]: We should be able to undo this change once Mux Core v7.0.5 is released and referenced in the Gradle file

File: MuxExoPlayer/src/main/java/com/mux/stats/sdk/muxstats/MuxBaseExoPlayer.java (hunk 1 above)
Comment (skidder) [reply MuxExoPlayer/src/main/java/com/mux/stats/sdk/muxstats/MuxBaseExoPlayer.java:192]: I agree, we should keep the constructor signature intact and ignore the parameter.

**Comment by daytime-em** [comment]
So Buildkite doesn't like this change for some reason, regardless of branch name or contributing user.

I've updated this PR with a simplification for the Demo and Test modules that allow them to depend directly on the`:MuxExoPlayer` directly. The automatic variant matching still seems to work in this context (try in the demos with exo versions 2.9.6/2.15.1/2.14.1 looking for log tag `RAAA`)

There's probably opportunity for more cleanup and simplification if this actually works with the tests, which are running now.

**Comment by daytime-em** [comment]
The recent changes fix the build but they change the dependency structure for demo and automatedtests.

Instead of pulling the exoplayer stats SDK from artifactory, it just depends on the project directly. Variant auto-matching seems sufficient to link the right variants for the demo and tests.

If the tests succeed, I think we should consider cleaning this up and taking it. We could maybe remove the fatAar variants entirely, though we should still probably deploy dev builds for testing in other client apps, etc

**Comment by skidder** [comment]
Closing, using #173 instead

**Pull request closed without merging**